### PR TITLE
Feature : Unsigned Txn Bundle Simulation

### PR DESCRIPTION
**Pull request by dmarzzz**
- This PR adds support from eth_callBundle RPC which was originally forked from flashbots MEV-GETH.
- Addition on top of mev-geth call bundle RPC are support for unsigned txns, payload modifications, and the addition of the geth internal struct logger to output JSON stack traces of EVM calls

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -2083,3 +2085,173 @@ func toHexSlice(b [][]byte) []string {
...
+func NewBundleAPI(b Backend, chain *core.BlockChain) *BundleAPI {
+	return &BundleAPI{b, chain}
+}
+
+// CallBundleArgs represents the arguments for a call.
+type CallBundleArgs struct {
+	Txs                    []TransactionArgs     `json:"txs"`
+	BlockNumber            rpc.BlockNumber       `json:"blockNumber"`
+	StateBlockNumberOrHash rpc.BlockNumberOrHash `json:"stateBlockNumber"`
+	Coinbase               *string               `json:"coinbase"`
```
Comment (axelKingsley) [inline internal/ethapi/api.go:2117]: Can you elaborate on the Coinbase argument?

File: core/tx_pool.go [hunk 2]
```diff
@@ -395,7 +395,7 @@ func (pool *TxPool) loop() {
 						pool.removeTx(tx.Hash(), true)
 					}
 					pool.dropTxFeed.Send(DropTxsEvent{
-						Txs: list,
+						Txs:    list,
```
Comment (axelKingsley) [inline core/tx_pool.go:398]: oh `go fmt` you sly dog...

File: core/state_processor.go [hunk 3]
```diff
@@ -155,3 +204,64 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
...
+			stack := make([]string, len(trace.Stack))
+			for i, stackValue := range trace.Stack {
+				stack[i] = stackValue.Hex()
+			}
+			formatted[index].Stack = &stack
+		}
+		if trace.Memory != nil {
+			memory := make([]string, 0, (len(trace.Memory)+31)/32)
+			for i := 0; i+32 <= len(trace.Memory); i += 32 {
+				memory = append(memory, fmt.Sprintf("%x", trace.Memory[i:i+32]))
```
Comment (axelKingsley) [inline core/state_processor.go:254]: A comment on exactly what's happening here would be a huge help in the future

File: core/state_processor.go [hunk 4]
```diff
@@ -155,3 +204,64 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, config, cfg)
 	return applyTransaction(msg, config, bc, author, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
 }
+
+func ApplyUnsignedTransactionWithResult(config *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, msg types.Message, usedGas *uint64, cfg vm.Config) (*types.Receipt, *ExecutionResult, []StructLogRes, error) {
+	// Create struct logger to get JSON stack traces
+	tracer := logger.NewStructLogger(nil)
```
Comment (axelKingsley) [inline core/state_processor.go:210]: might be a question for Louis, but creating a new logger every time this is called might be sub optimal?

**Review by axelKingsley — APPROVED** [review]
Just some nitpicks around commenting or clarity on things I didn't personally understand.

Also, a brief description of testing done in the PR would be helpful (you and I talked offline so I know this is tested, with more testing scheduled)

Looks awesome!

**Comment by dmarzzz** [comment]
Comments from this review will be addressed in the subsequent PR that will be made before taking to prod.

**Pull request merged**